tests/components/shelly/test_init.py

```diff
@@ -183,3 +187,27 @@ async def test_entry_unload_device_not_ready(
     await hass.async_block_till_done()
 
     assert entry.state is ConfigEntryState.NOT_LOADED
+
+
+async def test_entry_unload_firmware_upgrade(hass, mock_rpc_device, monkeypatch):
```

Comment:
```suggestion
async def test_entry_unload_not_connected(hass, mock_rpc_device, monkeypatch):
```
I think this would better represent what it does, it can also happen that we are disconnected due to other reason